File: lib/models/trip_settings.dart
class TripSettings {
  bool isDynamicTrip;
  bool includeFirstWalk;

  TripSettings(this.isDynamicTrip, this.includeFirstWalk);

  TripSettings.def()
      : isDynamicTrip = false,
        includeFirstWalk = false;

  Map<String, dynamic> toJson() {
    return {
      'isDynamicTrip': isDynamicTrip,
      'includeFirstWalk': includeFirstWalk
    };
  }

  bool equals(TripSettings other) {
    return isDynamicTrip == other.isDynamicTrip &&
        includeFirstWalk == other.includeFirstWalk;
  }

  factory TripSettings.fromJson(Map<String, dynamic> json) {
    return TripSettings(json['isDynamicTrip'], json['includeFirstWalk']);
  }

  factory TripSettings.from(TripSettings settings) {
    return TripSettings(settings.isDynamicTrip, settings.includeFirstWalk);
  }
}

File: android/app/src/main/java/com/example/retur/TripSettings.kt
package com.example.retur

data class TripSettings(
        val isDynamicTrip: Boolean,
        val includeFirstWalk: Boolean
) {
    companion object {
        fun fromJson(json: Map<String, Any>): TripSettings {
            return TripSettings(
                    json["isDynamicTrip"] as Boolean,
                    json["includeFirstWalk"] as Boolean
            )
        }
    }
}
